persepolis: 3.2.0 -> 4.0.1

iFreilicht opened the pull request:
Closes #288825


## Description of changes

Major version update from 3.2.0 to 4.0.1.

Auto-update didn't work because one patch didn't apply. Additionally, I could remove two patches. One has been merged upstream, the other one addressed an issue with building from source on darwin that was also fixed upstream.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

iFreilicht committed:
persepolis: 3.2.0 -> 4.0.0

Closes #288825

Auto-update didn't work because one patch didn't apply.
Additionally, I could remove one patch, as it has been merged upstream.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/3571

AndersonTorres commented on pkgs/tools/networking/persepolis/default.nix:
```diff
@@ -18,33 +18,26 @@
-    rev = version;
-    sha256 = "1rh7q432ynbysapsd075nif975ync71icpb71x2mb4j8jx1vzs45";
+    rev = "refs/tags/${version}";
+    sha256 = "sha256-2S6s/tWhI9RBFA26jkwxYTGeaok8S8zv/bY+Zr8TOak=";
```
```suggestion
    hash = "sha256-2S6s/tWhI9RBFA26jkwxYTGeaok8S8zv/bY+Zr8TOak=";
```

iFreilicht replied:
Ah yes, makes sense. Done!

AndersonTorres reviewed (commented):
Since the crappy github interface does not allow commenting non-diffed lines:

1. Reword description to be not self-referent: 

   `"A GUI for aria2"`

2. licenses should be gpl3Plus
3. Set meta.mainProgram
4. Migrate the package to by-name (optional but nice)

SuperSandro2000 commented on pkgs/tools/networking/persepolis/default.nix:
```diff
@@ -18,33 +18,26 @@
+    sha256 = "sha256-2S6s/tWhI9RBFA26jkwxYTGeaok8S8zv/bY+Zr8TOak=";
   };
 
   # see: https://github.com/persepolisdm/persepolis/blob/3.2.0/setup.py#L130
```
How are missing dependencies a problem here?

iFreilicht replied:
The issue was that the tests checked for the presence of some dependencies that weren't found. I now enabled the checkPhase again and instead fixed the tests themselves.

SuperSandro2000 commented on pkgs/tools/networking/persepolis/default.nix:
```diff
@@ -18,33 +18,26 @@
   doCheck = false;
 
-  preBuild=
   # Make setup automatic
```
That comment confused me more than it helps

iFreilicht replied:
I tried to improve it now, hope it helps.

SuperSandro2000 commented on pkgs/tools/networking/persepolis/default.nix:
```diff
@@ -18,33 +18,26 @@
-  preBuild=
   # Make setup automatic
-  ''
+  preBuild= ''
```
```suggestion
  preBuild = ''
```

iFreilicht replied:
Ah, good point. I actually ran nixpkgs-fmt on the whole file, forgot to do that before.

SuperSandro2000 commented on pkgs/tools/networking/persepolis/default.nix:
```diff
@@ -18,33 +18,26 @@
-        "import youtube_dl" "import yt_dlp as youtube_dl"
   '';
 
   patches = lib.optionals stdenv.isDarwin [
```
Can't we apply them even on linux so that linux users do not break them when updating the package?

iFreilicht replied:
Oh that's a good point! Indeed, the patches aren't very intrusive. Done!

iFreilicht committed:
persepolis: refactor: re-enable checkPhase

iFreilicht committed:
persepolis: formatting

Addressing review comments of
https://github.com/NixOS/nixpkgs/pull/292837

iFreilicht committed:
persepolis: apply Darwin patches on Linux as well

This will aid maintainability, as maintainers changing the package on
Linux will immediately see if the patches break without having to wait
for ofborg to tell them.

iFreilicht committed:
persepolis: fix incorrect meta fields

iFreilicht committed:
persepolis: migrate to by-name

iFreilicht commented:
@AndersonTorres I fixed all the meta attributes as suggested and migrated the package to by-name.

Thanks to you both for your review!

iFreilicht commented:
Fixed two small oversights in formatting, should be RTM now.

iFreilicht committed:
persepolis: 4.0.0 -> 4.0.1

See https://github.com/persepolisdm/persepolis/releases/tag/4.0.1
This version changed the build system from setuptools to meson.
It also fixed notifications on macOS when building from source.
This means that two patches could be deleted.

iFreilicht commented:
@SuperSandro2000 or @AndersonTorres, could you have a look again? I addressed all your comments and rebased on current master to resolve the conflict.

iFreilicht commented:
Actually, persepolis 4.0.1 was released in the meantime, so I added the required changes to update to that version as well.

mksafavi commented:
Is there any problem for this to get merged?

btw, version 4.1.0 is released too.
